### PR TITLE
feat(integrationconnectors): Prepared IntegrationConnectorsConnection for inclusion in C.G.C. documentation

### PR DESCRIPTION
**Pull request by balanaguharsha**
## Description

Fixes #<313761362>

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- Yes, **merge** this PR after it is approved
- [X] No, don't **merge** this PR after it is approved

**Style**

- My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [ Y] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ Y] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ Y] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by msampathkumar** [comment]
Hi @balanaguharsha thank you for the PR. Kindly also update https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS for this `integrationconnectors`. You can check `bigquery` CodeOwners entry for reference. CodeOwners are required for faster approvals.

**Comment by msampathkumar** [comment]
Sample failed CICD tests with following error

Error: Error creating Connection: googleapi: Error 400: The request was invalid: Service account 418598563416-compute@developer.gserviceaccount.com does not have permission to access the secret secret_version:"projects/418598563416/secrets/my-example-secrect/versions/1"

**Comment by balanaguharsha** [comment]
Need one more /gcbrun

**Comment by msampathkumar** [comment]
> Need one more /gcbrun

Unknown owner on line 23: make sure the team @terraform-google-modules/connectors-eng-team exists, is publicly visible, and has write access to the repository

/integrationconnectors/ @terraform-google-modules/connectors-eng-team @terraform-google-modules/terraform-samples-reviewers


When I cheked https://github.com/orgs/terraform-google-modules/teams/connectors-eng-team, it give me 404 Error

**Comment by msampathkumar** [comment]
Lint Error:

```
--- old/integrationconnectors/advanced/main.tf
+++ new/integrationconnectors/advanced/main.tf
@@ -252,6 +252,6 @@
     }
     enrichment_enabled = true
   }
-  depends_on        = [google_secret_manager_secret_iam_member.default]
+  depends_on = [google_secret_manager_secret_iam_member.default]
 }
```

File: integrationconnectors/advanced/main.tf
Comment (msampathkumar) [inline integrationconnectors/advanced/main.tf:21]: Please create a secret or as a last resort you can use test skip file to ignore tests for this sample.
```
Error: Error creating Connection: googleapi: Error 400: The request was invalid: Service account 711755423475-compute@developer.gserviceaccount.com does not have permission to access the secret secret_version:"projects/711755423475/secrets/my-example-secrect/versions/1"
```

File: .github/CODEOWNERS
Comment (balanaguharsha) [inline .github/CODEOWNERS:23]: Still gives the same error, please suggest.

**Comment by balanaguharsha** [comment]
PTAL.

File: .github/CODEOWNERS
Comment (msampathkumar) [reply .github/CODEOWNERS:23]: For this PR, please remove this line.

Use a new PR to update code-owners, after creating @terraform-google-modules/connectors-eng-team group.

**Review by msampathkumar — APPROVED** [review]

**Pull request merged**
